File: z-clients/javascript/src/models/metricOut.ts
// this file is @generated
import {
    type MetricType,
    MetricTypeSerializer,
} from './metricType';

export interface MetricOut {
    /** Label for this series */
    label: string;
    /** Human-readable description of this series */
    description: string;
    /** Key/Value pairs attached to this sequence */
    attributes: { [key: string]: string };
    /**
     * Most recent data point for this series
     * 
     * All points (u64, i64, and f64) are squished into an f64, be careful
     * of inexactness for values above 2**53.
     */
    value: number;
    /**
     * Type of this metric
     * 
     * Histograms are not currently exported through this API, and can
     * only be accessed through OTLP.
     */
    metricType: MetricType;
    /** Timestamp this metric was collected */
    timestamp: Date;
    /**
     * Optional unit, following UCUM unit conventions if possible
     * 
     * See https://ucum.org/ for details
     */
    unit?: string | null;
}

export const MetricOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MetricOut {
        return {
            label: object['label'],
            description: object['description'],
            attributes: object['attributes'],
            value: object['value'],
            metricType: MetricTypeSerializer._fromJsonObject(object['metric_type']),
            timestamp: new Date(Number(object['timestamp'])),
            unit: object['unit'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MetricOut): any {
        return {
            'label': self.label,
            'description': self.description,
            'attributes': self.attributes,
            'value': self.value,
            'metric_type': MetricTypeSerializer._toJsonObject(self.metricType),
            'timestamp': self.timestamp.getTime(),
            'unit': self.unit,
        };
    }
}

File: z-clients/rust/src/models/metric_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::metric_type::MetricType;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct MetricOut {
    /// Label for this series
    pub label: String,

    /// Human-readable description of this series
    pub description: String,

    /// Key/Value pairs attached to this sequence
    pub attributes: std::collections::HashMap<String, String>,

    /// Most recent data point for this series
    ///
    /// All points (u64, i64, and f64) are squished into an f64, be careful
    /// of inexactness for values above 2**53.
    pub value: f64,

    /// Type of this metric
    ///
    /// Histograms are not currently exported through this API, and can
    /// only be accessed through OTLP.
    pub metric_type: MetricType,

    /// Timestamp this metric was collected
    #[serde(with = "crate::unix_timestamp_ms_serde")]
    pub timestamp: jiff::Timestamp,

    /// Optional unit, following UCUM unit conventions if possible
    ///
    /// See https://ucum.org/ for details
    #[serde(skip_serializing_if = "Option::is_none")]
    pub unit: Option<String>,
}

impl MetricOut {
    pub fn new(
        label: String,
        description: String,
        attributes: std::collections::HashMap<String, String>,
        value: f64,
        metric_type: MetricType,
        timestamp: jiff::Timestamp,
    ) -> Self {
        Self {
            label,
            description,
            attributes,
            value,
            metric_type,
            timestamp,
            unit: None,
        }
    }

    pub fn with_unit(mut self, value: impl Into<Option<String>>) -> Self {
        self.unit = value.into();
        self
    }
}
